### PR TITLE
[CI:DOCS]first pass at adding swagger docs for api

### PR DESCRIPTION
**Pull request by baude**
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/api/server/register_containers.go [hunk 1]
```diff
@@ -10,42 +10,879 @@ import (
 )
 
 func (s *APIServer) RegisterContainersHandlers(r *mux.Router) error {
+	// swagger:operation GET /containers/create containers createContainer
```
Comment (jwhonce) [inline pkg/api/server/register_containers.go]: Wrong verb should be POST not GET

File: pkg/api/server/register_containers.go [hunk 2]
```diff
@@ -10,42 +10,879 @@ import (
...
+	//   '400':
+	//     description: bad parameter
+	//     schema:
+	//       "$ref": "#/types/ErrorModel"
+	//   '500':
+	//     description: unexpected error
+	//     schema:
+	//      "$ref": "#/types/ErrorModel"
 	r.HandleFunc(VersionedPath("/containers/json"), APIHandler(s.Context, generic.ListContainers)).Methods(http.MethodGet)
+	// swagger:operation GET /containers/prune containers pruneContainers
```
Comment (jwhonce) [inline pkg/api/server/register_containers.go]: Wrong verb

File: pkg/api/server/register_containers.go [hunk 3]
```diff
@@ -10,42 +10,879 @@ import (
...
+	// responses:
+	//   '200':
+	//     schema:
+	//       "$ref": "#/types/ContainerPruneReport"
+	//   '500':
+	//     description: unexpected error
+	//     schema:
+	//      "$ref": "#/types/ErrorModel"
 	r.HandleFunc(VersionedPath("/containers/prune"), APIHandler(s.Context, generic.PruneContainers)).Methods(http.MethodPost)
+	// swagger:operation DELETE /containers/json containers removeContainer
```
Comment (jwhonce) [inline pkg/api/server/register_containers.go]: Wrong path /containers/{name:..*}

File: pkg/api/server/register_containers.go [hunk 4]
```diff
@@ -10,42 +10,879 @@ import (
...
+	// parameters:
+	//  - in: path
+	//    name: nameOrID
+	//    required: true
+	//    description: the name or ID of the container
+	//  - in: query
+	//    name: detachKeys
+	//    type: string
+	//    description: needs description
+	// produces:
```
Comment (jwhonce) [inline pkg/api/server/register_containers.go]: Need to research what produces should include for empty body.

File: pkg/api/server/register_containers.go [hunk 5]
```diff
@@ -10,42 +10,879 @@ import (
...
+	//   '404':
+	//     description: no such container
+	//     schema:
+	//       "$ref": "#/types/ErrorModel"
+	//   '500':
+	//     description: unexpected error
+	//     schema:
+	//      "$ref": "#/types/ErrorModel"
 	r.HandleFunc(VersionedPath("/containers/{name:..*}/start"), APIHandler(s.Context, handlers.StartContainer)).Methods(http.MethodPost)
+	// swagger:operation GET /containers/{nameOrID}/start containers statsContainer
```
Comment (jwhonce) [inline pkg/api/server/register_containers.go]: Wrong path stats not start

File: pkg/api/server/register_containers.go [hunk 6]
```diff
@@ -10,42 +10,879 @@ import (
...
+	//   '404':
+	//     description: no such container
+	//     schema:
+	//       "$ref": "#/types/ErrorModel"
+	//   '500':
+	//     description: unexpected error
+	//     schema:
+	//      "$ref": "#/types/ErrorModel"
 	r.HandleFunc(VersionedPath("/containers/{name:..*}/stop"), APIHandler(s.Context, handlers.StopContainer)).Methods(http.MethodPost)
+	// swagger:operation POST /containers/{nameOrID}/top containers topContainer
```
Comment (jwhonce) [inline pkg/api/server/register_containers.go]: Wrong verb

File: pkg/api/server/register_containers.go [hunk 7]
```diff
@@ -10,42 +10,879 @@ import (
...
+	// ---
+	// parameters:
+	//  - in: path
+	//    name: nameOrID
+	//    required: true
+	//    description: the name or ID of the container
+	//  - in: query
+	//    name: signal
+	//    type: int
+	//    description: signal to be sent to container
```
Comment (jwhonce) [inline pkg/api/server/register_containers.go]: default: ... should be added to non-required parameters

File: pkg/api/server/register_containers.go [hunk 8]
```diff
@@ -10,42 +10,879 @@ import (
...
+	//       "$ref": "#/types/ErrorModel"
+	//   '409':
+	//     description: conflict
+	//     schema:
+	//       "$ref": "#/types/ErrorModel"
+	//   '500':
+	//     description: unexpected error
+	//     schema:
+	//      "$ref": "#/types/ErrorModel"
 	r.HandleFunc(VersionedPath("/libpod/containers/{name:..*}/kill"), APIHandler(s.Context, libpod.MountContainer)).Methods(http.MethodGet)
```
Comment (jwhonce) [inline pkg/api/server/register_containers.go]: The wrong function is being called here.

File: pkg/api/server/register_containers.go [hunk 9]
```diff
@@ -10,42 +10,879 @@ import (
...
+	// swagger:operation GET /libpod/containers/{nameOrID}/mount containers mountContainer
```
Comment (jwhonce) [inline pkg/api/server/register_containers.go]: An operation like mount should be a POST or PUT, not GET.

File: pkg/api/server/register_images.go [hunk 10]
```diff
@@ -10,32 +10,652 @@ import (
...
+	// - application/json
+	// parameters:
+	//  - in: query
+	//    name: fromImage
+	//    type: string
+	//    description: needs description
+	//  - in: query
+	//    name: tag
+	//    type: string
+	//    description: needs description
```
Comment (jwhonce) [inline pkg/api/server/register_images.go]: I suspect you also need a https://swagger.io/docs/specification/2-0/describing-request-body/ to complete this operation

File: pkg/api/server/register_images.go [hunk 11]
```diff
@@ -10,32 +10,652 @@ import (
...
+	//     schema:
+	//     items:
+	//       "$ref": "TBD"
+	//     description: no error
+	//   '500':
+	//     description: unexpected error
+	//     schema:
+	//      "$ref": "#/types/ErrorModel"
 	r.Handle(VersionedPath("/images/{name:..*}/get"), APIHandler(s.Context, generic.ExportImage)).Methods(http.MethodGet)
+	// swagger:operation GET /images/{nameOrID}/history images createImage
```
Comment (jwhonce) [inline pkg/api/server/register_images.go]: last field should be something like imageHistory

File: pkg/api/server/register_images.go [hunk 12]
```diff
@@ -10,32 +10,652 @@ import (
...
+	//     schema:
+	//       "$ref": "#/types/ErrorModel"
+	//   '500':
+	//     description: unexpected error
+	//     schema:
+	//      "$ref": "#/types/ErrorModel"
 	r.Handle(VersionedPath("/images/{name:..*}/tag"), APIHandler(s.Context, handlers.TagImage)).Methods(http.MethodPost)
-
-	// commit has a different endpoint
+	// swagger:operation POST /commit/ commit commitContainer
```
Comment (jwhonce) [inline pkg/api/server/register_images.go]: tag should be images not commit.

File: pkg/api/server/register_images.go [hunk 13]
```diff
@@ -10,32 +10,652 @@ import (
...
+	//    type: string
+	//    description: author of the image
+	//  - in: query
+	//    name: pause
+	//    type: bool
+	//    description: pause the container before committing it
+	//  - in: query
+	//    name: changes
+	//    type: string
+	//    description: instructions to apply while committing in Dockerfile format
```
Comment (jwhonce) [inline pkg/api/server/register_images.go]: Needs a requestBody stanza

File: pkg/api/server/register_images.go [hunk 14]
```diff
@@ -10,32 +10,652 @@ import (
...
+	//     description: no such image
+	//     schema:
+	//       "$ref": "#/types/ErrorModel"
+	//   '500':
+	//     description: unexpected error
+	//     schema:
+	//      "$ref": "#/types/ErrorModel"
 	r.Handle(VersionedPath("/libpod/images/{name:..*}/exists"), APIHandler(s.Context, libpod.ImageExists))
 	r.Handle(VersionedPath("/libpod/images/{name:..*}/tree"), APIHandler(s.Context, libpod.ImageTree))
+	// swagger:operation GET /libpod/images/{nameOrID}/history images createImage
```
Comment (jwhonce) [inline pkg/api/server/register_images.go]: operation should be imageHistory not createImage

File: pkg/api/server/register_containers.go (hunk 4 above)
Comment (baude) [reply pkg/api/server/register_containers.go]: agree, for next pr

File: pkg/api/server/register_images.go (hunk 12 above)
Comment (baude) [reply pkg/api/server/register_images.go]: it is actually a different endpoint. i think it should be commit

File: pkg/api/server/register_images.go (hunk 13 above)
Comment (baude) [reply pkg/api/server/register_images.go]: we dont actually support that, so i need to document that.

**Pull request merged**